### PR TITLE
Sort folders alphabetically in special folders screen

### PR DESCRIPTION
**Pull request by cketti**

### CODE REVIEW

File: feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/domain/usecase/GetSpecialFolderOptions.kt [hunk 1]
```diff
@@ -24,6 +24,10 @@ class GetSpecialFolderOptions(
                 ?: error("No incoming server settings available")
 
             val remoteFolders = folderFetcher.getFolders(serverSettings, authStateStorage)
+                .sortedWith(
+                    compareByDescending<RemoteFolder> { it.type == FolderType.INBOX }
+                        .thenBy(String.CASE_INSENSITIVE_ORDER) { it.displayName },
```
Comment (wmontwe) [inline feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/domain/usecase/GetSpecialFolderOptions.kt:29]: This moves the INBOX to the third position in the list of folders, why is that necessary?

Comment (cketti) [reply feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/domain/usecase/GetSpecialFolderOptions.kt:29]: It's done for consistency with the existing special folder preferences under *Settings → [Account] → Folders*. I believe we originally did this because that's also what we do in the folder list.

We can change the sort order later. Probably best to do it when we rewrite the settings screen. Then we can also share the logic to sort the list.

**Review by wmontwe — APPROVED** [review]
👍
